content/en/blog/_posts/2023-07-27-job-update-post.md

```diff
@@ -0,0 +1,207 @@
+---
+layout: blog
+title: "Kubernetes 1.28: Updates to the Job API"
+date: 2023-07-27
+slug: kubernetes-1-28-jobapi-update
+---
+
+**Authors:** Kevin Hannon (G-Research), Michał Woźniak (Google)
+
+This blog discusses two new features in Kubernetes 1.28 to improve Jobs for batch
+users: [PodReplacementPolicy](https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/3939-allow-replacement-when-fully-terminated)
+and [BackoffLimitPerIndex](https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/3850-backoff-limits-per-index-for-indexed-jobs).
+
+## Pod Replacement Policy
+
+### What problem does this solve?
+
+Currently, when a pod enters a terminating state (e.g. due to preemption or
+eviction), a replacement pod is created immediately, and both pods are running
+at the same time.
+
+This is problematic for some popular machine learning frameworks, such as
+TensorFlow and JAX, which require at most one pod running at the same time,
+for a given index (see more details in the [issue](https://github.com/kubernetes/kubernetes/issues/115844)).
+
+Creating the replacement Pod before the previous one fully terminates can also
+cause problems in clusters with scarce resources or with tight budgets. These
+resources can be difficult to obtain so pods can take a long time to find
+resources and they may only be able to find nodes until the existing pods are
+fully terminated. Further, if cluster autoscaler is enabled, the replacement
+Pods might produce undesired scale ups.
+
+### How can I use it
+
+This is an alpha feature, which you can enable by enabling the `JobPodReplacementPolicy`
+[feature gate](/docs/reference/command-line-tools-reference/feature-gates/) in
+your cluster.
+
+Once the feature is enabled you can use it by creating a new Job, which specifies
+`podReplacementPolicy` field as shown here:
+
+```yaml
+kind: Job
+metadata:
+  name: new
+  ...
+spec:
+  podReplacementPolicy: Failed
+  ...
+```
+
+Additionally, you can inspect the `.status.terminating` field of a Job. The value
+of the field is the number of Pods owned by the Job that are currently terminating.
+
+```shell
+kubectl get jobs/myjob -o yaml
+```
+
+```yaml
+apiVersion: batch/v1
+kind: Job
+status:
+  terminating: 3 # three Pods are terminating and have not yet reached the Failed phase
+```
+
+This can be particularly useful for external queueing controllers, such as
+[Kueue](https://github.com/kubernetes-sigs/kueue), that would calculate the
+quota and gate the start of a new Job until the resources are reclaimed from
+the currently terminating Job.
+
+### How can I learn more?
+
+- Read the KEP: [PodReplacementPolicy](https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/3939-allow-replacement-when-fully-terminated)
+
+## Job Backoff Limit per Index
+
+### What problem does this solve?
+
+Currently, for an [Indexed](/docs/concepts/workloads/controllers/job/#completion-mode)
```

Comment:
I find this sentence to be a bit awkward.

[Indexed Job]((/docs/concepts/workloads/controllers/job/#completion-mode)) failures are counted towards the global limit of retries.